### ISSUE TITLE
iOS Native crashes are not being reported to Sentry

### ISSUE THREAD

**Issue by MugiwaraxD** [getsentry/sentry-kotlin-multiplatform#305]
### Platform

Apple

### Installed

CocoaPods

### Version

0.10.0

### Steps to Reproduce

Failure Case
1. install sentry kmp 0.10.0
2. use shared initialization strategy
3. crash with nil exception on iOS
4. observe no crash report on sentry

Success case
1. install sentry kmp 0.10.0
2. use shared initialization strategy
3. crash with nil exception on KMP
4. observe crash report on sentry

### Expected Result

Expect to see crash report on Sentry from iOS.

### Actual Result

[sentry_crash_log.txt](https://github.com/user-attachments/files/18187487/sentry_crash_log.txt)
[sentry_crash_2nd_start.txt](https://github.com/user-attachments/files/18187488/sentry_crash_2nd_start.txt)

**Comment by buenaflor** [comment]
hey do you have more info on the setup?

is it possible to create a repro sample app?

testing it locally through our sample app works fine

**Comment by MugiwaraxD** [comment]
what's the best way to send a sample app? I have it reproduced

**Comment by buenaflor** [comment]
@MugiwaraxD if you like you can send me it zipped to giancarlo.buenaflor@sentry.io 

or even easier if you could open a github repo then I can directly try it out locally

**Comment by MugiwaraxD** [comment]
To further update this thread, the crash is showing when ran from AS for the test application and not showing in sentry when run directly from XCode

**Comment by buenaflor** [comment]
thx, that's an important detail

I've received the sample app, will look into it

**Comment by buenaflor** [comment]
I believe xcode is pausing the app processing when receiving a crash and therefore the SDK doesn't come to the point of storing the crash envelope.

The crash envelope is sent on the next app start. 

So essentially this happens because the crash occurs directly within init. I believe if you delay the crash within 5 seconds it would work also directly with running in xcode

If I just open the app directly on the simulator without xcode it works.

```
DispatchQueue.main.asyncAfter(deadline: .now() + 5.0) {
  let test: String? = nil
  let testString = test!
}
```

please correct me if I'm wrong

**Comment by MugiwaraxD** [comment]
If i just open the app directly on the simulator it does work with this block; however, if i run the app with the crash commented out, directly from XCode the crash will not be reported.

I'm also not sure if this is what is happening on my main project. I can send over some gradle files; however, I don't know how useful that is. In my main project I just tested that when running from Android Studio the native crash is still not sent, so sorry for misleading.

**Comment by buenaflor** [comment]
I cannot repro on my end unfortunately

Could you describe in detail how you debug this?

Something like

1. Start ios app in AS
2. Wait for crash
3. ... etc

**Comment by MugiwaraxD** [comment]
1. Start iOS App in XCode
2. Wait for crash
3. Crashes
4. Start iOS App in XCode with no crash in build
5. Does not crash + sentry does not report

**Comment by MugiwaraxD** [comment]
Here is a sample [https://zoox.sentry.io/issues/6153489478/?project=4508264377417728&query=is%3Aunreso[…]referrer=issue-stream&sort=date&statsPeriod=1h&stream_index=0](https://zoox.sentry.io/issues/6153489478/?project=4508264377417728&query=is%3Aunresolved%20issue.priority%3A%5Bhigh%2C%20medium%5D&referrer=issue-stream&sort=date&statsPeriod=1h&stream_index=0)

This crash was caught when running the iOS target from Android Studio. 

If i run the same thing but from XCode directly, it's not being reported

**Comment by buenaflor** [comment]
side note: I removed the link, I am not sure about the security aspect of posting that link publicly

this is the case that I described where xcode stops the app right after crashing. The SDK relies on storing the crash and if xcode stops the app before the SDK is able to store it then the crash won't be reported since there is none to report. 

I just confirmed this with `options.debug = true`, If I use xcode like you do then it says:`Sending 0 crash reports` which means the crash was never stored.

so I'd suggest to try out native crashes tied to some user interaction, e.g a button click and not directly after init.

It does work but just has this small edge case with xcode during debugging. in a normal prod app this is not a problem
